### PR TITLE
string_view: Add string_view::contains methods

### PR DESCRIPTION
**Pull request by Lastique**
These methods were added in C++23.

Closes https://github.com/boostorg/utility/issues/93.

@mclow Please take a look.

### CODE REVIEW

**Comment by glenfe** [comment]
It might be trivial, but I'd still prefer some unit tests.

**Comment by Lastique** [comment]
Agreed. Added tests and made a few corrections to the existing tests.

**Comment by mclow** [comment]
Im at C++Now at the moment ; I'll try to look at this, but it ,ay be next week

**Review by glenfe — APPROVED** [review]

File: test/string_ref_test2.cpp [hunk 1]
```diff
@@ -41,20 +44,19 @@ void ends_with ( const char *arg ) {
...
     string_ref sr  ( arg );
     string_ref sr2 ( arg );
-    const char *p = arg + std::strlen ( arg ) - 1;
-    while ( p >= arg ) {
-        std::string foo ( arg, p + 1 );
+    std::string foo ( arg );
+    while ( !foo.empty ()) {
         BOOST_TEST ( sr.starts_with ( foo ));
-        --p;
+        foo.erase ( foo.end () - 1 );
```
Comment (mclow) [inline test/string_ref_test2.cpp:59]: I think that `foo.erase(foo.rbegin());` would be more idiomatic.

File: test/string_view_test2.cpp [hunk 2]
```diff
@@ -68,12 +69,40 @@ void starts_with ( const char *arg ) {
...
 
+void contains ( const char *arg ) {
+    const std::size_t sz = std::strlen ( arg );
+    string_view sr  ( arg );
+    string_view sr2 ( arg );
+    std::string foo ( arg );
+    while ( !foo.empty ()) {
+        BOOST_TEST ( sr.contains ( foo ));
+        if ( ( foo.size () & 1u ) != 0u )
+            foo.erase ( foo.end () - 1 );
```
Comment (mclow) [inline test/string_view_test2.cpp:86]: Here even more so - `rbegin()`

**Review by mclow — COMMENTED** [review]
This looks OK to me; the additions are fine.  I have a couple of nits, and frankly, the `size_t` -> `std::size_t` changes were distracting.

Let me know what you think about `rbegin`

**Comment by Lastique** [comment]
> Let me know what you think about `rbegin`

`std::string` does not have `erase` by `reverse_iterator`. You would have to `foo.erase(foo.rbegin().base() - 1)`, which I don't think is any better.

**Comment by mclow** [comment]
> > Let me know what you think about `rbegin`
> 
> `std::string` does not have `erase` by `reverse_iterator`. 

How embarrassing.. My bad. I withdraw the suggestion.

**Comment by Lastique** [comment]
Ok, merging then. Thanks for the review.

**Pull request merged**